File: rb/features.rb
# FedoraMessaging SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module FedoraMessagingFeatures
  def self.make_feature(name)
    case name
    when "base"
      FedoraMessagingBaseFeature.new
    when "ratelimit"
      FedoraMessagingRatelimitFeature.new
    when "retry"
      FedoraMessagingRetryFeature.new
    when "test"
      FedoraMessagingTestFeature.new
    when "timeout"
      FedoraMessagingTimeoutFeature.new
    else
      FedoraMessagingBaseFeature.new
    end
  end
end

File: py/features.py
# FedoraMessaging SDK feature factory

from feature.base_feature import FedoraMessagingBaseFeature
from feature.test_feature import FedoraMessagingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: FedoraMessagingBaseFeature(),
        "test": lambda: FedoraMessagingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
